feature/experimental/bgp/ate_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go

```diff
@@ -0,0 +1,445 @@
+	}
+
+	// Check BGP globalAS from telemetry.
+	globalAS := gnmi.Get(t, dut, glblPath.State()).GetAs()
+	if globalAS != nbr.localAS {
+		t.Errorf("BGP globalAS: got %v, want %v", globalAS, nbr.localAS)
+	}
+
+	verifyPrefixesTelemetry(t, dut, 3, nbr.isV4)
+}
+
+func configureATE(t *testing.T, ateParams *bgpNbr, connectionType string, prefixes []string) *ondatra.ATETopology {
+	t.Helper()
+	t.Log("Configure ATE interface")
+	ate := ondatra.ATE(t, "ate")
+	port1 := ate.Port(t, "port1")
+	topo := ate.Topology().New()
+
+	iDut1 := topo.AddInterface(ateSrc.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateSrc.IPv4CIDR()).WithDefaultGateway(dutSrc.IPv4)
+	iDut1.IPv6().WithAddress(ateSrc.IPv6CIDR()).WithDefaultGateway(dutSrc.IPv6)
+
+	bgpDut1 := iDut1.BGP()
+	peer := bgpDut1.AddPeer().WithPeerAddress(ateParams.peerIP).WithLocalASN(ateParams.localAS)
+
+	if connectionType == connInternal {
+		peer.WithTypeInternal()
+	} else {
+		peer.WithTypeExternal()
+	}
+
+	if ateParams.isV4 {
+		network := iDut1.AddNetwork("bgpNeti1")
+		network.IPv4().WithAddress(prefixes[0]).WithCount(1)
+		network.BGP().WithNextHopAddress(ateSrc.IPv4).AddASPathSegment(55000, 4400, 3300)
+
+		network = iDut1.AddNetwork("bgpNeti2")
+		network.IPv4().WithAddress(prefixes[1]).WithCount(1)
+		network.BGP().WithNextHopAddress(ateSrc.IPv4).AddASPathSegment(55000, 7700)
+		network.BGP().WithNextHopAddress(ateSrc.IPv4).Communities().WithPrivateCommunities("200:1")
+
+		network = iDut1.AddNetwork("bgpNeti3")
+		network.IPv4().WithAddress(prefixes[2]).WithCount(1)
+		network.BGP().WithNextHopAddress(ateSrc.IPv4)
+	} else {
+		network := iDut1.AddNetwork("bgpNeti1")
+		network.IPv6().WithAddress(prefixes[0]).WithCount(1)
+		network.BGP().WithNextHopAddress(ateSrc.IPv6).AddASPathSegment(55000, 4400, 3300)
+
+		network = iDut1.AddNetwork("bgpNeti2")
+		network.IPv6().WithAddress(prefixes[1]).WithCount(1)
+		network.BGP().WithNextHopAddress(ateSrc.IPv6).AddASPathSegment(55000, 7700)
+		network.BGP().WithNextHopAddress(ateSrc.IPv6).Communities().WithPrivateCommunities("200:1")
+
+		network = iDut1.AddNetwork("bgpNeti3")
+		network.IPv6().WithAddress(prefixes[2]).WithCount(1)
+		network.BGP().WithNextHopAddress(ateSrc.IPv6)
+	}
+	return topo
+}
+
+func applyBgpPolicy(policyName string, dut *ondatra.DUTDevice) *oc.NetworkInstance_Protocol {
+	d := &oc.Root{}
+	ni1 := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	ni_proto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	bgp := ni_proto.GetOrCreateBgp()
+
+	pg := bgp.GetOrCreatePeerGroup("ATE")
+	pg.PeerGroupName = ygot.String("ATE")
+
+	pg.GetOrCreateApplyPolicy().ImportPolicy = []string{policyName}
+
+	return ni_proto
+}
+
+func createBgpNeighbor(nbr *bgpNbr, dut *ondatra.DUTDevice) *oc.NetworkInstance_Protocol {
+	d := &oc.Root{}
+	ni1 := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	ni_proto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
```

Comment:
no underscores in go variable names, only camelCase

Reply:
Addressed with latest commit